src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileLoggerFactory.cs

```diff
@@ -17,10 +17,10 @@ public LinuxAppServiceFileLoggerFactory()
             _logRootPath = Environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsLogsMountPath);
         }
 
-        public virtual LinuxAppServiceFileLogger GetOrCreate(string category)
+        public virtual LinuxAppServiceFileLogger GetOrCreate(string category, bool logBackoffEnabled = false)
         {
             return Loggers.GetOrAdd(category,
-                static (c, path) => new Lazy<LinuxAppServiceFileLogger>(() => new LinuxAppServiceFileLogger(c, path, new FileSystem())), _logRootPath).Value;
```

Comment:
The caching here is technically incorrect now. For example if we create 2 different loggers for category "foo" with logBackoffEnabled=true/false respectively, we'll cache the first and use it in all cases.

Reply:
I took note of this. The limitation of caching the logger is that hosting config for these settings only get propagated on host startup.  For fastlogging this matters less as it should only affect the first execution that occurs when a host starts its workers I believe.  This setting would only update on host start/restart. 
To avoid the state where the Host picks up a config change but, cannot apply it to the logger due to caching. I will try to make the FastLogging configuration static. Then whatever Host's state is should be applied to its logger when its instantiated and neither can be changed without a host restart. 

ExecutionDetails can make use of the HostingConfigMonitor. So in the case that detailed logs need to be shut off, we should be able to affect all running Functions.

Reply:
With this in mind, I chose to use the static definition of HostingConfiguration tied to lifecycle. There should hopefully be no confusion on the cached file logger now.  Whichever logger type is assigned to these events to begin with will persist.